### PR TITLE
fix(utils): add resize observer helper class

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Adds a `ResizeObserverHelper` util class to help reuse resize observer logic that's duplicated across multiple components 

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

File: src/components/KTruncate/KTruncate.vue [hunk 1]
```diff
@@ -228,24 +229,14 @@ const widthStyle = computed((): Record<string, string> => {
...
+
   resizeObserver.value.observe(kTruncateContainer.value as HTMLDivElement)
   updateToggleVisibility()
 })
 
 onBeforeUnmount(() => {
-  if (resizeObserver.value) {
-    resizeObserver.value.unobserve(kTruncateContainer.value)
-  }
+  resizeObserver.value?.unobserve(kTruncateContainer.value as HTMLDivElement)
```
Comment (adamdehaven) [inline src/components/KTruncate/KTruncate.vue:239]: issue: this should be done in `onUnmounted` instead:

> Use this hook to clean up manually created side effects such as timers, DOM event listeners or server connections.

[Source](https://vuejs.org/api/composition-api-lifecycle.html#onunmounted)

File: src/utilities/resizeObserverHelper.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/**
+ * ResizeObserverHelper
+ * @description Helper class for ResizeObserver
+ * @param callback - Callback function on resize
+ */
+
+export class ResizeObserverHelper {
```
Comment (adamdehaven) [inline src/utilities/resizeObserverHelper.ts:7]: question: why are we using a class instead of just a composable?

Comment (portikM) [reply src/utilities/resizeObserverHelper.ts:7]: So it looks close to native ResizeObserver API implementation. With native one you would do:
```
resizeObserver.value = new ResizeObserver()
resizeObserver.value.observe(...)
...
resizeObserver.value.unobserve(...)
```

With this helper wrapper it's only slightly different:
```
resizeObserver.value = ResizeObserverHelper.create()
resizeObserver.value.observe(...)
...
resizeObserver.value.unobserve(...)
```

Comment (portikM) [reply src/utilities/resizeObserverHelper.ts:27]: Done

**Pull request merged**